main/ble.c

```diff
@@ -442,10 +442,11 @@ int ble_foreach_characteristic(mac_addr_t mac,
     ble_device_t *dev;
     ble_service_t *service;
     ble_characteristic_t *characteristic;
+    ble_characteristic_t *characteristic_known;
 
     xSemaphoreTakeRecursive(devices_list_semaphore, portMAX_DELAY);
 
-    if (!(dev = ble_device_find_by_mac(devices_list, mac)))
+    if ((dev = ble_device_find_by_mac(devices_list, mac)) == NULL)
```

Comment:
I prefer the existing convention, please

Reply:
@shmuelzon  While you obviously have the last say about this, I think it makes sense to keep the code as simple and understandable as possible. If you look at the function implementation, it returns NULL. So, in my opinion it is more straightforward to compare it to null. While your solution is functionally the same, whoever is looking at this needs to consider that NULL is the same as false, which makes it slightly more complicated. There is just no direct correlation between what the function returns and this logic expression. What do you say?

Reply:
Okay, I still think that it should be desirable switching to a more readable and good-practice style.